catalog/transformer/catalog-transformer-pdf/src/main/java/ddf/catalog/transformer/input/pdf/PdfInputTransformer.java

```diff
@@ -81,9 +84,110 @@
     private static final FastDateFormat DATE_FORMAT = FastDateFormat.getInstance(
             "yyyy-MM-dd'T'HH:mm:ssZZ");
 
+    private final CheckedFunction<InputStream, PDDocument> inputStreamPDDocumentFunction;
+
+    private final CheckedFunction<PDDocument, String> geoParserFunction;
+
+    private final CheckedFunction<PDDocument, Optional<byte[]>> thumbnailFunction;
+
     private Map<ServiceReference, ContentMetadataExtractor> contentMetadataExtractors =
             Collections.synchronizedMap(new TreeMap<>(new ServiceComparator()));
 
+    private MetacardType metacardType;
+
+    private boolean usePdfTitleAsTitle;
+
+    /**
+     * @param metacardType       must be non-null
+     * @param usePdfTitleAsTitle must be non-null
+     */
+    public PdfInputTransformer(MetacardType metacardType, Boolean usePdfTitleAsTitle) {
+        this(metacardType,
+                usePdfTitleAsTitle,
+                PDDocument::load,
+                GEO_PDF_PARSER::getWktFromPDF,
+                PdfInputTransformer::generatePdfThumbnail);
```

Comment:
Just curious if you're using these method references for a particular reason besides unit testing? It definitely makes the code a little harder to follow when these are otherwise just normal methods calls. It is generally a code smell to have code changes purely for unit testing.